meta/splat.rb

```diff
@@ -0,0 +1,25 @@
+Mutest::Meta::Example.add :splat do
+  source 'foo(*bar)'
+
+  singleton_mutations
+  mutation 'foo'
+  mutation 'foo(nil)'
+  mutation 'foo(bar)'
+  mutation 'foo(self)'
+  mutation 'foo(*self)'
+end
+
+Mutest::Meta::Example.add :splat do
+  source 'a = *bar'
+
+  singleton_mutations
+  mutation 'a__mutest__ = *bar'
+  mutation 'a = nil'
+  mutation 'a = self'
+  mutation 'a = []'
+  mutation 'a = [nil]'
+  mutation 'a = [self]'
+  mutation 'a = [*self]'
+  mutation 'a = [bar]'
+  mutation s(:lvasgn, :a, s(:splat, s(:send, nil, :bar)))
```

Comment:
It's slightly tricky because the original expression parses as

```
(lvasgn :a
  (array
    (splat
      (send nil :bar))))
```

We could check in the `array` mutator if replacing the `array` node with its contents would leave a splat in an invalid place, but I'm not exactly sure which cases to control for there.